From eada89cecf90196b0ae3a247170a123c3f63ff50 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:01:17 -0400
Subject: [usb] core: fix incorrect type in assignment in usb_set_lpm_parameters()

Message-id: <1410887463-20674-6-git-send-email-dzickus@redhat.com>
Patchwork-id: 95558
O-Subject: [RHEL7 PATCH 005/591] usbcore: fix incorrect type in assignment in usb_set_lpm_parameters()
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 4d96799524159f7c3e774a7b45a308780a26dc4d
Author: Xenia Ragiadakou <burzalodowa@gmail.com>
Date:   Sat Aug 31 18:09:13 2013 +0300

    usbcore: fix incorrect type in assignment in usb_set_lpm_parameters()

    In the bos usb_ss_cap_descriptor structure, bU2DevExitLat is of type __le16.
    This value is used as it is, without being first converted to the CPU
    byteorder, for the setup of usb device's usb3_lpm_parameters.
    This patch fixes that by converting bU2DevExitLat field to the CPU byteorder
    before the assignmenment to [udev/hub]_u2_del variables.

    Signed-off-by: Xenia Ragiadakou <burzalodowa@gmail.com>
    Reported-by: kbuild test robot <fengguang.wu@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/hub.c b/drivers/usb/core/hub.c
index 879651c..a0210f4 100644
--- a/drivers/usb/core/hub.c
+++ b/drivers/usb/core/hub.c
@@ -315,9 +315,9 @@ static void usb_set_lpm_parameters(struct usb_device *udev)
   return;
 
  udev_u1_del = udev->bos->ss_cap->bU1devExitLat;
- udev_u2_del = udev->bos->ss_cap->bU2DevExitLat;
+ udev_u2_del = le16_to_cpu(udev->bos->ss_cap->bU2DevExitLat);
  hub_u1_del = udev->parent->bos->ss_cap->bU1devExitLat;
- hub_u2_del = udev->parent->bos->ss_cap->bU2DevExitLat;
+ hub_u2_del = le16_to_cpu(udev->parent->bos->ss_cap->bU2DevExitLat);
 
  usb_set_lpm_mel(udev, &udev->u1_params, udev_u1_del,
    hub, &udev->parent->u1_params, hub_u1_del);
-- 
1.7.1